activerecord/test/cases/finder_test.rb

```diff
@@ -265,6 +265,11 @@ def test_find_by_sql_with_sti_on_joined_table
     assert_equal [Account], accounts.collect(&:class).uniq
   end
 
+  def test_find_by_association_subquery
+    author = authors(:david)
+    assert_equal author.post, Post.find_by(author_id: Author.where(id: author))
```

Comment:
can you also add a test for `assert_equal author.post, Post.find_by(author: Author.where(id: author))` ?
Btw, is that case working right now, or this patch fix that too?